openeogeotrellis/job_tracker.py

```diff
@@ -131,8 +131,8 @@ def update_statuses(self) -> None:
                                         result_metadata["usage"] = usage
                                     registry.patch(job_id, user_id, **result_metadata)
 
-                                    registry.mark_done(job_id, user_id)
-                                    _log.info("marked %s as done" % job_id, extra={'job_id': job_id})
+                                    # TODO: this `set_status` is just here to trigger the "mark_done from status" logic. Can this be automated further?
+                                    registry.set_status(job_id, user_id, new_status)
```

Comment:
There's also a status change a few lines earlier (with `patch`) so this doesn't feel just right/needs more work.

Reply:
indeed, the status change above is done with `patch` instead of `set_status` 
and the former does not support this "auto-mark-done" feature (hence the TODO note).

The problem with adding auto-mark-done directly to `patch`, is that it changes the zookeeper path of the batch job which potentially messes up subsequent job_registry operations (e.g. `remove_dependencies` stuff following on `patch`). I wanted to keep this first refactor step simple

Reply:
Yes, I believe most/all of those mutators only work on "ongoing" nodes so you want to be careful to not move them to the "done" branch too early.

But as a first step, this is fine by me.